blob/azureblob/azureblob.go

```diff
@@ -318,8 +318,34 @@ var errUnimplemented = errors.New("not implemented")
 
 // Copy implements driver.Copy.
 func (b *bucket) Copy(ctx context.Context, dstKey, srcKey string, opts *driver.CopyOptions) error {
-	// TODO(rvangent): Implement this and delete errUnimplemented.
-	return errUnimplemented
+	dstKey = escapeKey(dstKey, false)
+	dstBlobURL := b.containerURL.NewBlobURL(dstKey)
+	srcKey = escapeKey(srcKey, false)
+	srcURL := b.containerURL.NewBlobURL(srcKey).URL()
+	resp, err := dstBlobURL.StartCopyFromURL(ctx, srcURL, nil /* metadata */, azblob.ModifiedAccessConditions{}, azblob.BlobAccessConditions{})
+	if err != nil {
+		return err
+	}
+	copyStatus := resp.CopyStatus()
+	nErrors := 0
+	for copyStatus == azblob.CopyStatusPending {
+		// Poll until the copy is complete.
+		time.Sleep(500 * time.Millisecond)
```

Comment:
I see copy in general may take more time. Is it common enough that we might expose an option for timeouts? Also, maybe we should do exponential waits, or maybe it's too much.

Reply:
I've never seen it actually return `CopyStatusPending`, so I suspect it's only for very large files. We can expose an option for the 500ms if we're asked to do so, I don't think it's worth it at this point.